mmcv/cnn/rfsearch/search.py

```diff
@@ -0,0 +1,249 @@
+        self.by_epoch = by_epoch
+
+    def init_model(self, model: nn.Module):
+        """init model with search ability.
+
+        Args:
+            model (nn.Module): pytorch model
+
+        Raises:
+            NotImplementedError: only support three modes:
+                search/fixed_single_branch/fixed_multi_branch
+        """
+        if self.verbose:
+            logger.info('RFSearch init begin.')
+        if self.mode == 'search':
+            if self.config['structure']:
+                self.set_model(model, search_op='Conv2d')
+            self.wrap_model(model, search_op='Conv2d')
+        elif self.mode == 'fixed_single_branch':
+            self.set_model(model, search_op='Conv2d')
+        elif self.mode == 'fixed_multi_branch':
+            self.set_model(model, search_op='Conv2d')
+            self.wrap_model(model, search_op='Conv2d')
+        else:
+            raise NotImplementedError
+        if self.verbose:
+            logger.info('RFSearch init end.')
+
+    def after_train_epoch(self, runner):
+        """Performs a dilation searching step after one training epoch."""
+        if self.by_epoch and self.mode == 'search':
+            self.step(runner.model, runner.work_dir)
+
+    def after_train_iter(self, runner):
+        """Performs a dilation searching step after one training iteration."""
+        if not self.by_epoch and self.mode == 'search':
+            self.step(runner.model, runner.work_dir)
+
+    def step(self, model: nn.Module, work_dir: str):
+        """Performs a dilation searching step.
+
+        Args:
+            model (nn.Module): pytorch model
+            work_dir (str): Directory to save the searching results.
+        """
+        self.config['search']['step'] += 1
+        if (self.config['search']['step']
+            ) % self.config['search']['search_interval'] == 0 and (self.config[
+                'search']['step']) < self.config['search']['max_step']:
+            self.search(model)
+            for name, module in model.named_modules():
+                if isinstance(module, ConvRFSearchOp):
+                    self.config['structure'][name] = module.op_layer.dilation
+
+            write_to_json(
+                self.config,
+                os.path.join(
+                    work_dir,
+                    'local_search_config_step%d.json' %
+                    self.config['search']['step'],
+                ),
+            )
+        elif (self.config['search']['step'] +
+              1) == self.config['search']['max_step']:
+            self.search_estimate_only(model)
+
+    def search(self, model: nn.Module):
+        """estimate and search for RFConvOp.
+
+        Args:
+            model (nn.Module): pytorch model
+        """
+        for module in model.modules():
+            if isinstance(module, ConvRFSearchOp):
+                module.estimate()
+                module.expand()
+
+    def search_estimate_only(self, model):
```

Comment:
please add type hints and docstring for this method.

Reply:
In fact this function is never used, so I have deleted it.

Reply:
OK